components/gantry/singleaxis/singleaxis.go

```diff
@@ -57,10 +57,13 @@ func (cfg *Config) Validate(path string) ([]string, error) {
 		return nil, errors.Wrap(err, "mm_per_rev must be non-zero and positive")
 	}
 
-	if len(cfg.Board) == 0 && len(cfg.LimitSwitchPins) > 0 {
+	if cfg.Board == "" && len(cfg.LimitSwitchPins) > 0 {
 		return nil, errors.New("gantries with limit_pins require a board to sense limit hits")
 	}
-	deps = append(deps, cfg.Board)
+
+	if cfg.Board != "" {
```

Comment:
only append the dependency if it exists, that was the error with the dependency not ready message.